mmv1/templates/terraform/constants/artifact_registry_repository.go.erb

```diff
@@ -0,0 +1,54 @@
+<%# The license inside this block applies to this file
+  # Copyright 2023 Google Inc.
+  # Licensed under the Apache License, Version 2.0 (the "License");
+  # you may not use this file except in compliance with the License.
+  # You may obtain a copy of the License at
+  #
+  #     http://www.apache.org/licenses/LICENSE-2.0
+  #
+  # Unless required by applicable law or agreed to in writing, software
+  # distributed under the License is distributed on an "AS IS" BASIS,
+  # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  # See the License for the specific language governing permissions and
+  # limitations under the License.
+-%>
+func upstreamPoliciesDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
+  log.Printf("%v, %v, %v\n", k, old, new)
```

Comment:
Unless we have a reason to do this by hand, it would probably be better to use the sdk provided set methods such as [schema.NewSet](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/set.go#L63-L70) and [Set.Equal](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/set.go#L157-L163) to ensure we don't miss anything with casting the types around.

Reply:
Agreed! Switched over to using schema.Set.